File: QuasarCode_Library_Maths/Units/Quantities.cs
using System;
using System.Collections.Generic;
using System.Text;

using QuasarCode.Library.Maths.Units.Common;

namespace QuasarCode.Library.Maths.Units
{
    /// <summary>
    /// Physical quantitys that have units
    /// </summary>
    public enum Quantities
    {
        /// <summary>No assigned quantity</summary>
        None,

        /// <summary>Mesurement of rotation</summary>
        Angle,

        /// <summary>Mesurement of space</summary>
        Length,

        /// <summary>Mesurement of matter</summary>
        Mass,

        /// <summary>Mesurement of change</summary>
        Time,

        /// <summary>Mesurement of electricity</summary>
        ElectricCurrent,

        /// <summary>Mesurement of thermal energy</summary>
        Temperature,

        /// <summary>Mesurement of quantity</summary>
        Quantity,

        /// <summary>Mesurement of change</summary>
        LuminousIntensity
    }

    public abstract class QuantityBase : IQuantity
    {
        public string Name { get; }

        protected QuantityBase(string name)
        {
            this.Name = name;
        }
    }

    public sealed class NoneQuantity : QuantityBase
    {
        public NoneQuantity() : base("None") { }
    }

    public sealed class Angle : QuantityBase
    {
        public Angle() : base("Angle") { }
    }

    public sealed class Length : QuantityBase
    {
        public Length() : base("Length") { }
    }

    public sealed class Mass : QuantityBase
    {
        public Mass() : base("Mass") { }
    }

    public sealed class Time : QuantityBase
    {
        public Time() : base("Time") { }
    }

    public sealed class ElectricCurrent : QuantityBase
    {
        public ElectricCurrent() : base("ElectricCurrent") { }
    }

    public sealed class Temperature : QuantityBase
    {
        public Temperature() : base("Temperature") { }
    }

    public sealed class Quantity : QuantityBase
    {
        public Quantity() : base("Quantity") { }
    }

    public sealed class LuminousIntensity : QuantityBase
    {
        public LuminousIntensity() : base("LuminousIntensity") { }
    }
}

File: QuasarCode_Python/QuasarCode/Maths/Units/Quantities.py
from enum import Enum
from QuasarCode.Maths.Units._IQuantity import IQuantity

class Quantities(Enum):
    """
    Physical quantitys that have units

    none - No assigned quantity
    Angle - Mesurement of rotation
    Length - Mesurement of space
    Mass - Mesurement of matter
    Time - Mesurement of change
    ElectricCurrent - Mesurement of electricity
    Temperature - Mesurement of thermal energy
    Quantity - Mesurement of quantity
    LuminousIntensity - Mesurement of electromagnitism
    """

    # No assigned quantity
    none,

    # Mesurement of rotation
    Angle,

    # Mesurement of space
    Length,

    # Mesurement of matter
    Mass,

    # Mesurement of change
    Time,

    # Mesurement of electricity
    ElectricCurrent,

    # Mesurement of thermal energy
    Temperature,

    # Mesurement of quantity
    Quantity,

    # Mesurement of electromagnitism
    LuminousIntensity


class QuantityBase(IQuantity):
    def __init__(self, name):
        self.__Name = name

    def getName(self):
        return self.__Name


class NoneQuantity(QuantityBase):
    def __init__(self):
        super().__init__("None")

class Angle(QuantityBase):
    def __init__(self):
        super().__init__("Angle")

class Length(QuantityBase):
    def __init__(self):
        super().__init__("Length")

class Mass(QuantityBase):
    def __init__(self):
        super().__init__("Mass")

class Time(QuantityBase):
    def __init__(self):
        super().__init__("Time")

class ElectricCurrent(QuantityBase):
    def __init__(self):
        super().__init__("ElectricCurrent")

class Temperature(QuantityBase):
    def __init__(self):
        super().__init__("Temperature")

class Quantity(QuantityBase):
    def __init__(self):
        super().__init__("Quantity")

class LuminousIntensity(QuantityBase):
    def __init__(self):
        super().__init__("LuminousIntensity")